xml/System.IO/FileSystemWatcher.xml

```diff
@@ -823,8 +823,33 @@
   
  The <xref:System.IO.FileSystemWatcher.WaitForChanged%2A> method allows event handlers to be invoked to respond to file changes even if this property is set to `false`.  
   
-   
-  
+ If an <xref:System.IO.IOException> is thrown
```

Comment:
It would be good to mention why the limits are there, and how to persist the setting.

Suggestion (to be improved):

On Linux the number of watchers and directories that can be watched is limited. These limits protect the amount of kernel memory that is available for watchers. When a limit is reached `FileSystemWatcher` will throw an `IOException` or report it via the `FileSystemWatcher.Error` event. The limits can be read from `/proc/sys/fs/inotify/max_user_instances` and `/proc/sys/fs/inotify/max_user_watches`.

The files can be used to set the kernel values, but the changes will not persist across reboots. On most distributions, you can persist the settings by adding the following to `/etc/sysctl.conf`.
```
fs.inotify.max_user_instances=<value>
fs.inotify.max_user_watches=<value>
```
As values you can use double of the values reported by the `proc` files. After changing the file, you can run `sysctl -p` to apply the settings.